src/Swarm/Language/Syntax.hs

```diff
@@ -774,9 +785,20 @@ mkOp c s1@(Syntax l1 _) s2@(Syntax l2 _) = Syntax newLoc newTerm
   sop = noLoc (TConst c)
   newTerm = SApp (Syntax l1 $ SApp sop s1) s2
 
--- | The surface syntax for the language
-data Syntax = Syntax {sLoc :: SrcLoc, sTerm :: Term}
-  deriving (Eq, Show, Data, Generic, FromJSON, ToJSON)
+-- | The surface syntax for the language, with location and type annotations.
+data Syntax' ty = Syntax'
+  { _sLoc :: SrcLoc
+  , _sTerm :: Term' ty
+  , _sType :: ty
+  }
+  deriving (Eq, Show, Functor, Foldable, Traversable, Data, Generic, FromJSON, ToJSON)
+
+type Syntax = Syntax' ()
+
+pattern Syntax :: SrcLoc -> Term -> Syntax
+pattern Syntax l t = Syntax' l t ()
+
+{-# COMPLETE Syntax #-}
```

Comment:
What does `COMPLETE` do?

Reply:
It specifies a set of constructors/pattern synonyms which should be considered "complete" for the purposes of pattern match coverage checking.  i.e. if you pattern-match on everything listed in the `COMPLETE` pragma then GHC will not warn about missing cases.  Using `COMPLETE` is common when creating pattern synonyms since there's no way in general to know whether an arbitrary set of pattern synonyms is going to cover all possible cases, so without the pragma in general GHC will generate lots of warnings about incomplete pattern matches.

https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/pragmas.html#complete-pragmas